controllers/openstacknet_controller.go

```diff
@@ -102,6 +92,52 @@ func (r *OpenStackNetReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
+	//
+	// initialize condition
+	//
+	cond := &ospdirectorv1beta1.Condition{}
+
+	//
+	// If RoleReservations status map is nil, create it
+	//
+	if instance.Status.RoleReservations == nil {
+		instance.Status.RoleReservations = map[string]ospdirectorv1beta1.OpenStackNetRoleStatus{}
+	}
+
+	//
+	// Used in comparisons below to determine whether a status update is actually needed
+	//
+	currentStatus := instance.Status.DeepCopy()
+	statusChanged := func() bool {
+		return !equality.Semantic.DeepEqual(
+			r.getNormalizedStatus(&instance.Status),
+			r.getNormalizedStatus(currentStatus),
+		)
+	}
+
+	defer func(cond *ospdirectorv1beta1.Condition) {
+		//
+		// Update object conditions
+		//
+		instance.Status.CurrentState = ospdirectorv1beta1.NetState(cond.Type)
+		instance.Status.Conditions.UpdateCurrentCondition(
+			cond.Type,
+			ospdirectorv1beta1.ConditionReason(cond.Message),
+			cond.Message,
+		)
+
+		if statusChanged() {
+			if updateErr := r.Client.Status().Update(context.Background(), instance); updateErr != nil {
+				if err == nil {
+					err = common.WrapErrorForObject(
+						"Update Status", instance, updateErr)
+				} else {
+					common.LogErrorForObject(r, updateErr, "Update status", instance)
+				}
+			}
+		}
+	}(cond)
```

Comment:
Nice.  Is this a standard pattern for handling Conditions or did you create this yourself?

Reply:
no, have not found this to be a standard pattern. in cnosp we used the defer to update the status, I enhanced this to manage the condition in that way